hoa2dot/parsers.py

```diff
@@ -173,29 +174,39 @@ def body(self, args):
     def state_name(self, args):
         """Parse the 'state_name' node."""
         non_trees = [arg for arg in args if not isinstance(arg, Tree)]
-        kwargs = {arg.data: arg.children[0] for arg in args if isinstance(arg, Tree)}
+        # kwargs = {arg.data: arg.children[0] for arg in args if isinstance(arg, Tree)}
 
-        if len(args) == 1:
+        # make acc_sig a set not a list, this is naive...
+        kwargs = dict()
+        for arg in args:
+            if isinstance(arg, Tree):
+                if arg.data == "acc_sig":
+                    kwargs[arg.data] = set()
+                    kwargs[arg.data].add(arg.children[0])
+                else:
+                    kwargs[arg.data] = arg.children[0]
+
```

Comment:
I would change the above code in the following way.

`kwargs` contains `acc_sig` as key, if it is specified. So if `acc_sig` is present in `kwargs` dictionary, just make it as set.

```python
    def state_name(self, args):
        """Parse the 'state_name' node."""
        non_trees = [arg for arg in args if not isinstance(arg, Tree)]
        kwargs = {arg.data: arg.children[0] for arg in args if isinstance(arg, Tree)}
        # as before.
        # this changes:
        if "acc_sig" in kwargs:
            kwargs["acc_sig"] = set(kwargs["acc_sig"])
```

untested.

Reply:
yes, that's a better way to change it